net/dns/resolver/forwarder.go

```diff
@@ -916,10 +920,7 @@ func (f *forwarder) forwardWithDestChan(ctx context.Context, query packet, respo
 		metricDNSFwdDropBonjour.Add(1)
 		res, err := nxDomainResponse(query)
 		if err != nil {
-			f.logf("error parsing bonjour query: %v", err)
```

Comment:
discussed: in theory this is unreachable, as if the query failed to parse we should get this far. the comment is also wrong - returning an error is actually handled by mdnsresponder, but per #13512 it may be picky about the error structure

Reply:
Additionally, if this were reachable, this code would deadlock waiting to receive from the `responses` channel on line 334:
https://github.com/tailscale/tailscale/blob/f3de4e96a8e2ba191816d89ce8fbe02ebb572bb1/net/dns/resolver/tsdns.go#L322-L334
And this code would panic for the same reason:
https://github.com/tailscale/tailscale/blob/f3de4e96a8e2ba191816d89ce8fbe02ebb572bb1/net/dns/resolver/tsdns.go#L419-L433